execution/src/eth_token.rs

```diff
@@ -231,6 +231,48 @@ pub fn transfer_eth<G: GasMeter>(
     Ok(())
 }
 
+pub fn replicate_transfers<G: GasMeter, L: EthTransferLog>(
+    eth_transfer_logger: &L,
+    session: &mut Session,
+    traversal_context: &mut TraversalContext,
+    gas_meter: &mut G,
+) -> Result<(), moved_shared::error::Error> {
+    // Transfer the transaction value from EVM native account to `origin`.
+    // This step is needed because all EVM transactions start with the caller
+    // transferring tokens to the EVM native account as part of `evm_call`.
```

Comment:
I was a bit confused about this phrasing and it took me a while to figure out what's happening. Maybe it'd help to specify that this happens as part of the deposited tx execution and we design all mints by default to happen to the EVM native address.

Reply:
Additional comment added in https://github.com/MovedNetwork/op-move/pull/319/commits/2ac0015e5aa4e8fc63257354212957ee28c723a6